boards: nucleo_g474re: Use HSE instead of HSI

alexanderwachter opened the pull request:
Change the board defconfig to use the provided 24MHz Crystal instead
of the internal resonator.
This improves the clock precision and stability.

FRASTM commented on boards/arm/nucleo_g474re/nucleo_g474re_defconfig:
```diff
@@ -27,16 +27,17 @@ CONFIG_CLOCK_CONTROL=y
+CONFIG_CLOCK_STM32_PLL_SRC_HSE=y
+CONFIG_CLOCK_STM32_HSE_CLOCK=24000000
 
 CONFIG_CLOCK_STM32_PLL_M_DIVISOR=4
```
Proposal for an aditionnal comment to produce the PLLCLK at 170MHz, the combination would be
`CONFIG_CLOCK_STM32_PLL_M_DIVISOR=6`
with a 
`CONFIG_CLOCK_STM32_PLL_N_MULTIPLIER=85`

alexanderwachter replied:
Done.

FRASTM commented:
This change could also apply to nucleo_g431rb board as well

alexanderwachter committed:
boards: nucleo_g431rb: Use HSE instead of HSI

Change the board defconfig to use the provided 24MHz Crystal instead
of the internal resonator.
This improves the clock precision and stability.

Signed-off-by: Alexander Wachter <alexander@wachter.cloud>

alexanderwachter committed:
boards: nucleo_g474re: Use HSE instead of HSI

Change the board defconfig to use the provided 24MHz Crystal instead
of the internal resonator.
This improves the clock precision and stability.

Signed-off-by: Alexander Wachter <alexander@wachter.cloud>

FRASTM commented:
Frequency is set correctly on nucleo_g431rb with HSE for PLLCLK at 150MHz

erwango commented:
@alexanderwachter, since it was in first, I propose you rebase on https://github.com/zephyrproject-rtos/zephyr/pull/24157, then we'll merge this one

erwango commented:
@alexanderwachter can you rebase ?

alexanderwachter commented:
@FRASTM The q output from the PLL should actually generate 48MHz right? Otherwise it is useless for USB.
